src/commands/commit/upload.ts

```diff
@@ -30,32 +32,50 @@ export class UploadCommand extends Command {
 
   public repositoryURL?: string
 
-  private apiKeyValidator: ApiKeyValidator
+  private cliVersion: string
   private config = {
     apiKey: process.env.DATADOG_API_KEY,
   }
   private dryRun = false
 
   constructor() {
     super()
-    this.apiKeyValidator = new ApiKeyValidator(this.config.apiKey)
+    this.cliVersion = require('../../../package.json').version
   }
 
   public async execute() {
     const initialTime = new Date().getTime()
-    const api = this.getApiHelper()
     this.context.stdout.write(renderCommandInfo(this.dryRun))
 
-    const cliVersion = require('../../../package.json').version
-    const metricsLogger = getMetricsLogger(cliVersion)
-    const payload = await this.getPayloadToUpload(cliVersion)
-    if (!payload) {
+    const metricsLogger = getMetricsLogger(this.cliVersion)
+    const apiKeyValidator = new ApiKeyValidator(this.config.apiKey, datadogSite, metricsLogger.logger)
+    const payload = await getCommitInfo(await newSimpleGit(), this.context.stdout, this.repositoryURL)
+    if (payload === undefined) {
       return 0
     }
     try {
-      await this.uploadRepository(api, metricsLogger.logger, payload)
+      const requestBuilder = this.getRequestBuilder()
+      const status = await this.uploadRepository(requestBuilder)(payload, {
+        apiKeyValidator,
+        onError: (e) => {
+          this.context.stdout.write(renderFailedUpload(e.message)), metricsLogger.logger.increment('failed', 1)
```

Comment:
is this correct?
Shouldn't it be:
```suggestion
          this.context.stdout.write(renderFailedUpload(e.message))
          metricsLogger.logger.increment('failed', 1)
```